pkg/controllers/resources/ingresses/util/util.go

```diff
@@ -80,12 +83,13 @@ func ProcessAlbAnnotations(namespace string, k string, v string) (string, string
 }
 
 func UpdateAnnotations(vObj client.Object) client.Object {
-	annotations := vObj.GetAnnotations()
-	for k, v := range annotations {
-		delete(annotations, k)
+	// we need to do that otherwise the original object will be modified
+	ingress := *vObj.(*networkingv1.Ingress)
```

Comment:
I don't think this a good practice, if you want to clone the object, do `DeepCopy()`, also why is modifying the object now an issue?

Reply:
because the underlying map gets modified, and we didn't care about it before because there was just the host object being modified at that point. With the new defer stuff, if the map is modified the virtual annotations also get modified